### PR TITLE
[16.0][MIG] repair_discount: Migration to 16.0

### PR DESCRIPTION
**Pull request by unaiberis**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
/ocabot migration repair_discount

Check the tests.

**Review by anajuaristi — APPROVED** [review]

File: repair_discount/models/mrp_repair.py [hunk 1]
```diff
@@ -9,94 +9,50 @@
...
         for line in self:
-            taxes = line.tax_id.compute_all(
-                line.price_unit,
-                line.repair_id.pricelist_id.currency_id,
-                line.product_uom_qty,
-                line.product_id,
-                line.repair_id.partner_id,
-            )
-            line.price_total = taxes["total_included"] * (
+            line.price_total = line.price_total * (1 - (line.discount or 0.0) / 100.0)
```
Comment (pedrobaeza) [inline repair_discount/models/mrp_repair.py]: Calculate `1 - line.discount / 100.0` once and use it for both. The `or 0.0` is useless.

File: repair_discount/models/mrp_repair.py [hunk 2]
```diff
@@ -112,20 +68,20 @@ def _create_invoices(self, group=False):
                 fee_lines.invoice_line_id.with_context(
                     check_move_validity=False
                 ).update({"discount": fee_lines.discount})
-        self.invoice_id.with_context(
-            check_move_validity=False
-        )._recompute_dynamic_lines(
-            recompute_all_taxes=True, recompute_tax_base_amount=True
-        )
+
```
Comment (pedrobaeza) [inline repair_discount/models/mrp_repair.py]: Don't add empty lines in the middle of a method.

File: repair_discount/models/mrp_repair.py [hunk 3]
```diff
@@ -112,20 +68,20 @@ def _create_invoices(self, group=False):
...
         return res
 
     def _calculate_line_base_price(self, line):
         return line.price_unit * (1 - (line.discount or 0.0) / 100.0)
 
+    amount_tax = fields.Monetary(
```
Comment (pedrobaeza) [inline repair_discount/models/mrp_repair.py]: Are you redefining this field as computed?

File: repair_discount/tests/test_mrp_repair_discount.py [hunk 4]
```diff
@@ -1,13 +1,19 @@
...
-class TestMrpRepairDiscount(common.SavepointCase):
+_logger = logging.getLogger(__name__)
+
+
+class TestMrpRepairDiscount(TransactionCase):
     @classmethod
     def setUpClass(cls):
-        super(TestMrpRepairDiscount, cls).setUpClass()
+        super().setUpClass()
+
```
Comment (pedrobaeza) [inline repair_discount/tests/test_mrp_repair_discount.py:11]: Don't add empty lines inside a method (this applies to all the occurrences).

File: repair_discount/tests/test_mrp_repair_discount.py [hunk 5]
```diff
@@ -54,7 +68,9 @@ def setUpClass(cls):
                 "discount": 50,
             }
         )
+        _logger.debug("Created repair line with discount: %s", cls.repair_line.discount)
```
Comment (pedrobaeza) [inline repair_discount/tests/test_mrp_repair_discount.py]: Remove this debug entries. It doesn't bring value.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

File: repair_discount/models/mrp_repair.py (hunk 3 above)
Comment (unaiberis) [reply repair_discount/models/mrp_repair.py]: Yes, you are right, this is wrong

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**